src/soroban.js

```diff
@@ -0,0 +1,74 @@
+import BigNumber from 'bignumber.js';
+
+/**
+ * Soroban helper class
+ * formatting, parsing, and etc
+ * @class Soroban
+ */
+export class Soroban {
+  /**
+   * format token amount to its ideal specific number of decimals
+   *
+   * @param {BigNumber} amount - the token amount you want to display
```

Comment:
can we make this `bigint | number | string`, instead?

Reply:
even if we need to use BigNumber _internally_, let's make them pass in a native value

Reply:
doesn't it need to be a BigInt or BigNumber since the number can be bigger than MAX_INT?

Reply:
I'm saying we should have these be the signatures:

```typescript
function parseTokenAmount(value: string | number | bigint, decimals: number): string;
function formatTokenAmount(value: string, decimals: number): bigint;
```

Then, you can `let bigNum = BigNumber(value)` internally to work with it.

If they want to parse a floating point value that exceeds the limits of `number`, they can use a `string` (even if that means `BigNumber(value).toString()`, but then it's _their_ dependency, not ours :laughing:)

Reply:
ahhhh ya I see I see. Thanks for explaining, yeah that sounds like an easy way to avoid the dep.

Reply:
hmm but we use `BigNumber` for calculation in `parseTokenAmount`
`const wholeValue = new BigNumber(whole);
const fractionValue = new BigNumber(fraction);

return wholeValue.shiftedBy(decimals).plus(fractionValue);`

Reply:
deleted my comment because I see the other comment mentioning it